biz.aQute.repository/src/aQute/maven/provider/MetadataParser.java

```diff
@@ -108,9 +108,12 @@ public Tag toTag() {
 			Tag versioning = new Tag(top, "versioning");
 
 			Tag snapshot = new Tag(versioning, "snapshot");
```

Comment:
I think you need a test to not emit the snapshot tag if all of the fields are null. Now you can end up with an empty snapshot element.

Reply:
got it, the two sets are supposed to be kind of mutually exclusive.